### ISSUE TITLE
{Stream, LazyList}.iterate(_)(_).take(_).size is toooooooooo slow (exponentially) in Scala 2.13.x

### ISSUE THREAD

**Issue by xuwei-k** [scala/bug#10872]
```
Welcome to Scala 2.13.0-M4-pre-20d3c21 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_172).
Type in expressions for evaluation. Or try :help.

scala> Stream.iterate(0){ a => println(a) ; a + 1 }.take(7).size
       ^
       warning: value Stream in package scala is deprecated (since 2.13.0): Use LazyList instead of Stream
0
0
1
0
1
2
0
1
2
3
0
1
2
3
4
0
1
2
3
4
5
0
1
2
3
4
5
6
res0: Int = 7

scala> LazyList.iterate(0){ a => println(a) ; a + 1 }.take(7).size
0
0
1
0
1
2
0
1
2
3
0
1
2
3
4
0
1
2
3
4
5
res1: Int = 7
```

```
Welcome to Scala 2.12.6 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_172).
Type in expressions for evaluation. Or try :help.

scala> Stream.iterate(0){ a => println(a) ; a + 1 }.take(7).size
0
1
2
3
4
5
res0: Int = 7
```

**Comment by julienrf** [comment]
I think this has been fixed by https://github.com/scala/scala/pull/6495

**Comment by xuwei-k** [comment]
> I think this has been fixed by scala/scala#6495

Oh, sorry 🙇
